src/Dapr.Actors/Runtime/ActorRegistrationCollection.cs

```diff
@@ -36,5 +36,21 @@ public void RegisterActor<TActor>(Action<ActorRegistration> configure = null)
             configure?.Invoke(registration);
             this.Add(registration);
         }
+
+        /// <summary>
+        /// Registers an actor type in the collection.
+        /// </summary>
+        /// <typeparam name="TActor">Type of actor.</typeparam>
+        /// <param name="actorTypeName">The name of the actor type represented by the actor.</param>
+        /// <param name="configure">An optional delegate used to configure the actor registration.</param>
+        /// <remarks>The value of <paramref name="actorTypeName"/> will have precedence over the default actor type name derived from the actor implementation type or any type name set via <see cref="ActorAttribute"/>.</remarks>
+        public void RegisterActor<TActor>(string actorTypeName, Action<ActorRegistration> configure = null)
+            where TActor : Actor
+        {
+            var actorTypeInfo = ActorTypeInformation.Get(typeof(TActor), actorTypeName);
```

Comment:
It's going to take me a bit to review this and rationalize the change. I have to check whether there are other places where we call `ActorTypeInformation.Get`.

Reply:
Ok. To clarify, this should be a non-breaking change. I added an overload for `ActorTypeInformation.Get` which takes an string parameter for specifying the actor type name.

```csharp
public static ActorTypeInformation Get(Type actorType)
``` 

The above still exists, but will call the below with `actorTypeName: null`.
```csharp
public static ActorTypeInformation Get(Type actorType, string actorTypeName)
```

And then in the above method I've just modified it to check is `actorTypeName` is provided.

Reply:
> Ok. To clarify, this should be a non-breaking change. I added an overload for ActorTypeInformation.Get which takes an string parameter for specifying the actor type name.

What I'm concerned about isn't the addition of a new method it's a the introduction of a 'split brain' problem. We need to make sure it's not possible for two parts of the system to disagree about the type name.

Reply:
Can you please do the following:

- On ActorTypeInformation - mark `TryGet(Type)` and `Get(Type)` as `[Obsolete]`- we want to make sure we don't have code calling those since they don't accept a type name override.
- On ActorHost we need another overload of `CreateForTest` that accepts a type name.
- Update the remaining references to `Get(Type)` to call `GetType(Type, null)` - these will be unit tests

Reply:
Fixed this here https://github.com/bjorkstromm/dotnet-sdk/commit/e1b62c9ddcde129c269a77bd477bcaac717f0820

But since this PR was unintentionally closed, changes won't appear here. Could you please reopen, or should I open a new PR?